src/commands/play.rs

```diff
@@ -138,12 +216,18 @@ async fn calculate_time_until_play(queue: &[TrackHandle], mode: PlayMode) -> Opt
     }
 }
 
-async fn enqueue_playlist(call: &Arc<Mutex<Call>>, uri: &str, mode: PlayMode) {
+async fn enqueue_playlist(
+    call: &Arc<Mutex<Call>>,
+    uri: &str,
+    mode: PlayMode,
+) -> Option<Vec<String>> {
     if let Some(urls) = YouTubeRestartable::ytdl_playlist(uri, mode).await {
-        for url in urls {
-            enqueue_song(call, url.to_string(), true, mode).await;
+        for url in urls.iter() {
```

Comment:
Why did we add the `.iter()` here?

Reply:
Because `urls` is now returned and the loop was taking ownership.